### PR TITLE
Add examples to debugging docs

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Partially stolen from @drashna on discord and #5498, I have added some useful examples on how to debug after enabling debug, as well as mentioning hid_listen as an alternative to QMK Toolbox.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/newbs_testing_debugging.md [hunk 1]
```diff
@@ -27,6 +27,14 @@ void keyboard_post_init_user(void) {
 }
 ```
 
+### Debugging With QMK Toolbox
+
+For compatible platforms, [QMK Toolbox](https://github.com/qmk/qmk_toolbox) can be used to display debug messages from your keyboard.
+
+### Debugging With hid_listen
+
+Prefer a terminal based solution? [hid_listen](https://www.pjrc.com/teensy/hid_listen.html), provided by PJRC, can also be used to display deboug messages. Prebuilt binaries for Windows,Linux,and MacOS are available.
```
Comment (ezuk) [inline docs/newbs_testing_debugging.md]: `debug` (typo)

File: docs/newbs_testing_debugging.md [hunk 2]
```diff
@@ -41,3 +49,52 @@ After that you can use a few different print functions:
 * `uprintf("%s string", var)`: Print a formatted string
 * `dprint("string")` Print a simple string, but only when debug mode is enabled
 * `dprintf("%s string", var)`: Print a formatted string, but only when debug mode is enabled
+
+## Debug Examples
+
+Collection of real world debugging examples. For additional information, refer to [Debugging/Troubleshooting QMK](faq_debug.md).
```
Comment (ezuk) [inline docs/newbs_testing_debugging.md]: `Below is a collection`

File: docs/newbs_testing_debugging.md (hunk 1 above)
Comment (zvecr) [reply docs/newbs_testing_debugging.md]: Good spot, fixed.

File: docs/newbs_testing_debugging.md (hunk 2 above)
Comment (zvecr) [reply docs/newbs_testing_debugging.md]: Fixed.

**Comment by Lenbok** [comment]
One thing I have been confused about is all the printf varieties. E g. Some examples here use xprintf but that isn't in the list given above. Could you add an explanation about that?

**Comment by drashna** [comment]
"it's complicated".

It changes how/where printing happens...
If you check /tmk_core/common/print.h, that changes how/what is defined.

**Comment by zvecr** [comment]
@Lenbok I would be happy to do a 2nd pass with more info once its been collated, but i would rather keep the scope of this PR to examples I have actually used, and are confident on its implementation.

**Comment by Lenbok** [comment]
I think it's making things more confusing in a document aimed at newbs to be adding a key logging example making use of a `xprintf` function that wasn't in the set of ones described literally only a few lines above. Is there a reason why that example doesn't use the `uprintf` function listed? It sounds like you're not confident that `uprintf` would work - if you want I'll test that tonight, and then maybe we can just update that example to be consistent.

File: docs/newbs_testing_debugging.md [hunk 3]
```diff
@@ -41,3 +49,52 @@ After that you can use a few different print functions:
...
+### Which matrix position is this keypress?
+
+When porting, or when attempting to diagnose pcb issues, it can be useful to know if a keypress is scanned correctly. To enable logging for this scenario, add the following code to your keymaps `keymap.c`
+
+```c
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+  // If console is enabled, it will print the matrix position and status of each key pressed
+#ifdef CONSOLE_ENABLE
+    xprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed);
```
Comment (Lenbok) [inline docs/newbs_testing_debugging.md]: I just tested, and this example works fine if you change `xprintf` to `uprintf` (or `dprintf` for the version that only starts outputting once debug is enabled).

**Comment by zvecr** [comment]
I was not my confidence in `xprintf` vs `uprintf`, more that i did not want to add docs for `xprintf` when i had also not heard of it. That being said, i totally agree that `uprintf` is more consistent and have also updated and verified its behaviour.

**Comment by zvecr** [comment]
Anything i can do to push this over the finish line?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**